src/FastTags/LexerM.hs

```diff
@@ -0,0 +1,780 @@
+----------------------------------------------------------------------------
+-- |
+-- Module      :  FastTags.LexerTypes
+-- Copyright   :  (c) Sergey Vinokurov 2019
+----------------------------------------------------------------------------
+
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE DeriveFunctor              #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase                 #-}
+{-# LANGUAGE MagicHash                  #-}
+{-# LANGUAGE MultiWayIf                 #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE UnboxedTuples              #-}
+
+{-# OPTIONS_GHC -O2 #-}
```

Comment:
It seems confusing to override optimization flags... the cabal file is the usual place to put things like -O2.  I assume it's because it makes a special difference here but is not worth it for other modules?  If -O2 doesn't actually hurt other modules but significantly helps this one, how about putting it in the cabal file with a comment.  Or if it really should be just this file then document the line above.

Reply:
Makes sense, I'll move it to the cabal file.